torch/csrc/jit/codegen/cuda/index_idgraph_utils.cpp

```diff
@@ -0,0 +1,686 @@
+#include <torch/csrc/jit/codegen/cuda/contiguity.h>
+#include <torch/csrc/jit/codegen/cuda/index_compute.h>
+#include <torch/csrc/jit/codegen/cuda/index_idgraph_utils.h>
+#include <torch/csrc/jit/codegen/cuda/index_reference_replay.h>
+#include <torch/csrc/jit/codegen/cuda/ir_utils.h>
+#include <torch/csrc/jit/codegen/cuda/lower2device.h>
+#include <torch/csrc/jit/codegen/cuda/lower_utils.h>
+#include <torch/csrc/jit/codegen/cuda/lower_validation.h>
+
+namespace torch {
+namespace jit {
+namespace fuser {
+namespace cuda {
+
+namespace {
+
+//! A struct to keep track of necessary parameters used in
+//!  configuring index compute pass.
+//! These should be the parameters that we need to populate
+//!  from the actual tv and loop nest and would not be able
+//!  to read directly from idgraph.
+//! TODO:
+//!   Would expect this list to become shorter over time,
+//!  as more info can be determined holistically.
+struct IndexingParameters {
+  //! Initial binding of index math to concrete iterdomain ids,
+  //!  from the loop nest analysis.
+  std::unordered_map<IterDomain*, Val*> initial_concrete_id_index;
+
+  //! (Used in non-global indexing) the concrete iterdomains that
+  //!  we want to skip or merge into contiguous indexing paths.
+  std::unordered_set<IterDomain*> zero_domains;
+
+  //! (Used in non-global indexing) the preferred path we would
+  //!  be propagating contiguously merged indices backward.
+  std::unordered_set<IterDomain*> preferred_concrete_ids;
+
+  //! The inferred halo padded extents of the concrete iterdomains.
+  std::unordered_map<IterDomain*, Val*> concrete_id_to_halo_extent;
```

Comment:
Yes, eventually. The halo extent is currently still loop nest dependent, as is anything in LoopIndex class in general. so that'd need to be fixed before we can have it in compute at map. 

Will address in a follow up.